### ISSUE TITLE
can't install groovy-eclipse and web-developer tools at the same time?

### ISSUE THREAD

**Issue by jeacott1** [groovy/groovy-eclipse#1286]
I don't seem to be able to install groovy-eclipse and web-developer tools at the same time.
anyone got any hints?

![image](https://user-images.githubusercontent.com/16600100/127077552-06f0d7e3-e05c-4745-8043-5632f837af1a.png)

**Comment by eric-milles** [comment]
Can you try to install each individually?  Chances are only one makes reference to `logback-classic`.

**Comment by eric-milles** [comment]
If you are using eclipse 2020-09, 2020-12 or 2021-06, can you retry with the snapshot update site?

**Comment by jeacott1** [comment]
yeah I've tried quite a few variations now, they all ended up like this. 
but after clicking around a bunch the ui let me hit the 'next' button and installed things anyway ¯\_(ツ)_/¯ .
I seem to have somehow also installed the m2e integration too.
unfortunately I cant see any option to select groovy 2.5 anywhere. previously I'd have had both 2.5 and 3 as options in the groovy settings panel. what am I missing?

![image](https://user-images.githubusercontent.com/16600100/127244382-1979bba2-1bf2-476c-ba9a-375d34bf8cf4.png)

global config
![no options](https://user-images.githubusercontent.com/16600100/127244458-0184878c-1a91-471d-9fec-7561db6c7f7d.JPG)

project config
![project_option](https://user-images.githubusercontent.com/16600100/127244461-a01e1dc5-0e6d-4457-bb4f-d3bac761b61f.JPG)

**Comment by eric-milles** [comment]
Can you start over with https://dist.springsource.org/snapshot/GRECLIPSE/e4.20

**Comment by jeacott1** [comment]
@eric-milles thanks for the assist,
I upgraded with https://dist.springsource.org/snapshot/GRECLIPSE/e4.20
no conflicts during the upgrade this time but after the restart eclipse 'forgot' that about 200 loaded modules were maven so I spent an hour wrangling them back to mvn projects. no clue if its related to this update or was just eclipse being eclipse.
still no groovy compiler options except 'I don't care' though :(

![image](https://user-images.githubusercontent.com/16600100/127410501-4a032064-dfd0-4f6e-a374-04e2b348aa51.png)

**Comment by eric-milles** [comment]
@jeacott1 I can't tell from your last comment if you started from a fresh install or tried to update the broken one.  It is the `org.codehaus.groovy.eclipse.compilerResolver` that fills in the drop-down of compiler versions.  You can check if you have it installed at _Help > About Eclipse ...> Installation Details > Plug-ins_.  If you start your eclipse with the `-console` command line switch, there will be some output from it like this:
```
Starting Groovy-Eclipse compiler resolver. Specified compiler level: 2.5
 56 org.codehaus.groovy_2.5.14.202107271059 STARTING
```
You can also look in the _Error Log_ view to see if there were any issues reported at startup.  Probably best to clear it with the red X and restart to see only the errors for a single startup.

**Comment by jeacott1** [comment]
@eric-milles  - I had upgraded, not started fresh.
so today I started fresh, and with a fresh workspace and yay, I got the groovy 2.5 button back in the global config!
however, the moment I switched to my usual project workspace I was back to only having groovy 3 and no other options.
unfortunately thereafter even if I switch back to the previous workspace the config option is gone :(

now here's the thing I've discovered, and I'm not sure if its a groovy-eclipse bug, or an eclipse bug.
easy to replicate though.
If you 'switch workspaces', ie open a workspace, the file->switch workspace. then groovy-eclipse is always broken.
however, if I shut down eclipse and open it fresh from the desktop groovy-eclipse works as expected!
This happens even with fresh empty workspaces from a brand new eclipse install (I'm using the self contained version of eclipse where everything is unzipped under the eclipse folder, not the one that installs and scatters all the dependencies everywhere - I've had very little success with the that deployment of eclipse) 
the only changes I've made to the stock install are to add the groovy-eclipse snapshot site and install. and I added a local jdk8 compiler.

I hope that helps find the cause of this issue, but for now I have a viable workaround. 
thank you for your help!

**Comment by jeacott1** [comment]
@eric-milles just fyi, I updated the eclipse-groovy tools etc today and was back to not having any options.
 after the installation restart I also had an almost empty package explorer. 
closing and starting eclipse again recovered my workspace, but alas groovy was back to "I don't care" as the only option available.
a further shutdown and restart has recovered my groovy options. I have no clue why restarts no longer work. 
the language server for xml also seems sticky and weird. I've also disabled that now too. 
![image](https://user-images.githubusercontent.com/16600100/128114911-e375d08b-a63b-4fdb-9064-81c26a1fb2aa.png)
